### PR TITLE
mac: fix explicit symbol exports

### PR DESCRIPTION
**Pull request by Riolku**
Exception information necessary for properly catching exceptions, seems to not be exported on Mac unless we export the exception, which seems to be different from Linux & Windows.

In additional, Mac issues a warning when building the Python module (which itself builds as a shared library). The issue is that the python module links with our static library, and the static library does not export symbols (since it doesn't do anything for static libraries).

However, this mismatch in visibility causes a warning. Technically, we shouldn't build a shared library with a static library since the static library isn't position-independent code, but we don't really have other options unless we embed all of the object files?

My current solution is to keep the visibility annotations in the static library, which seems to work fine and avoids the warning. However, we should probably find some other solution. Embedding all of the object files isn't super appealing from a design perspective but seems the easiest. It'd be great to embed the shared library.. but that would require some changes to how we package for python, and I haven't looked into it.

Ref #2136.

### CODE REVIEW

**Comment by Riolku** [comment]
Well, my solution seems to break windows.

**Comment by Riolku** [comment]
After some thought, the best solution is probably to link the python library against `libkuzu.so`

File: src/include/common/api.h [hunk 1]
```diff
@@ -13,9 +13,6 @@
 #define KUZU_HELPER_DEPRECATED __attribute__((__deprecated__))
 #endif
 
-#ifdef KUZU_STATIC_DEFINE
-#define KUZU_API
```
[the remark this answers is not in the document]
Comment (Riolku) [reply src/include/common/api.h:17]: Yeah that makes sense. I'm primarily wondering how best to avoid the Mac warning and the Windows failure.

**Comment by Riolku** [comment]
Actually I'll just ship this without fixing the warning for now and defer this to post release.

**Review by benjaminwinger — APPROVED** [review]

**Pull request merged**